Xamarin.Essentials/Screenshot/Screenshot.macos.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using ImageIO;
+using MobileCoreServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        internal static bool PlatformIsCaptureSupported => true;
+
+        const string appServicesPath =
+    "/System/Library/Frameworks/ApplicationServices.framework/Versions/Current/ApplicationServices";
+
+        // https://developer.apple.com/documentation/coregraphics/1454595-cgdisplaycreateimage
+        [DllImport(appServicesPath, EntryPoint = "CGDisplayCreateImage")]
+        static extern /* CGImageRef */ IntPtr CGDisplayCreateImage(int displayId);
+
+        static Task<ScreenshotResult> PlatformCaptureAsync()
+        {
+            using var pool = new NSAutoreleasePool();
+            var location = NSApplication.SharedApplication.KeyWindow.Frame.Location;
+            var screens = new List<NSScreen>(NSScreen.Screens);
+            var screen = screens.First(obj => obj.Frame.Contains(location));
```

Comment:
This might thro a confusing exception if no screen was found. Maybe do a FirstOrDefault and then throw an invalid op ex with a useful message... 

This also brings up a though for me. Is it possible that a view can be off all the screens? What are the cases where this could actually fail? Does macOS support having no screens?

Reply:
>Does macOS support having no screens?

Good question... I had `FirstOrDefault()`, but then I thought its not possible to have.
But it could be:
You have a mac mini, start the app, unplug the screen and click...
OK, i will catch that 🥳